### PR TITLE
#1162 Age Diagnoses Card Data fetch/transform

### PR DESCRIPTION
**Pull request by ciaranschutte**

### CODE REVIEW

File: src/components/CohortBuilder/Summary/AgeDiagChart.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    colors={[theme.chartColors.lightblue]}
+  />
+);
+
+export const ageDiagQuery = sqon => ({
+  query: `query($sqon:JSON) {
+    participant {
+      aggregations(filters: $sqon) {
+        diagnoses__age_at_event_days {
+          histogram(interval: 1){
```
Comment (hlminh2000) [inline src/components/CohortBuilder/Summary/AgeDiagChart.js]: I think an interval of `365` and further bucketing based on years would relieve a lot of CPU cycles in the client :)

Comment (ciaranschutte) [reply src/components/CohortBuilder/Summary/AgeDiagChart.js]: @hlminh2000 good shout!

File: src/components/CohortBuilder/Summary/AgeDiagChart.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+              key    
+            }
+          } 
+        }
+      
+      } 
+    }
+  }
+  `,
+  variables: sqon,
```
Comment (hlminh2000) [inline src/components/CohortBuilder/Summary/AgeDiagChart.js]: Oops, I think you meant `variables: { sqon }`

File: src/components/CohortBuilder/Summary/AgeDiagChart.js (hunk 1 above)
Comment (ciaranschutte) [reply src/components/CohortBuilder/Summary/AgeDiagChart.js]: @hlminh2000 364? 0 - 364 for newborn etc

Comment (jberube) [reply src/components/CohortBuilder/Summary/AgeDiagChart.js]: Depends on the year ;)

Comment (jberube) [reply src/components/CohortBuilder/Summary/AgeDiagChart.js]: Might be of interest, or not:
https://pumas.jpl.nasa.gov/files/04_21_97_1.pdf

File: src/components/CohortBuilder/Summary/AgeDiagChart.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+  transform: data => {
+    const buckets = get(
+      data,
+      'data.participant.aggregations.diagnoses__age_at_event_days.histogram.buckets',
+    );
+
+    // NB: intervals are in days
+    const reducer = (prev, current) => prev + current.doc_count;
```
Comment (jberube) [inline src/components/CohortBuilder/Summary/AgeDiagChart.js]: `sum` would be a better name

Comment (jberube) [reply src/components/CohortBuilder/Summary/AgeDiagChart.js]: Or using lodash `sumBy` https://lodash.com/docs/4.17.11#sumBy

**Pull request merged**